### PR TITLE
Filter results from disabled orgunits.

### LINKED ISSUE #3273: Orgunit's Inactive flag not respected by the new responsible fields
The new vocabulary sources used in ITask.responsible etc. (OGIP 15), provides also entries for inactive OrgUnits. Users from an inactive OrgUnit should newer be selectable.

### PR DESCRIPTION
**Pull request by maethu**
The problem was introduced with the OGIP 15 (https://github.com/4teamwork/opengever.core/pull/2852), which turned all vocabularies (IVocabulary) into sources (ISource) to support a new kind of selection widget provided by ftw.keywordwidget.

With this change filtering for disabled OrgUnits went lost.
This PR brings back this feature.

I also slightly refactored the source a little bit, otherwise this change would introduced a lot of duplicated code. 

I also hat to fix a side effect on the advanced search form.
A test-odgs user was not properly configured, since a OrgUnit was missing.
This change basically no longer allows user without an OrgUnit, since it is required to build the query.

Fixes #3273

### CODE REVIEW

**Comment by maethu** [comment]
I'm struggling, sry...

Seems like my change has some more implications than I thought.

**Comment by maethu** [comment]
@4teamwork/gever 
Well, I kicked the job again an it passes now 🤔 
I guess this is now ready to review

File: opengever/ogds/base/sources.py [hunk 1]
```diff
@@ -40,6 +40,7 @@ def base_query(self):
         query = create_session().query(User, OrgUnit) \
                                 .join(OrgUnit.users_group) \
                                 .join(Group.users)
+        query = query.filter(OrgUnit.enabled)
```
Comment (lukasgraf) [inline opengever/ogds/base/sources.py]: This relies on implicit conversion of this expresion to boolean.

[`Query.filter()`](http://docs.sqlalchemy.org/en/latest/orm/query.html#sqlalchemy.orm.query.Query.filter) doesn't show an example of a single-value expression without comparison operators - so I think SQLAlchemy is magically doing `bool(OrgUnit.enabled) == True` behind the scenes for your case.

But I'd rather see an explicit `filter(OrgUnit.enabled == True)` here (and the other comparisons), relying on SQLAlchemy expression magic too much can go wrong and cause very inefficient SQL queries.

Comment (maethu) [inline opengever/ogds/base/sources.py]: @lukasgraf thx good catch, was not aware of that. 👍

**Pull request merged**